pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -103,7 +102,15 @@
     private final BatchMessageContainerBase batchMessageContainer;
     private CompletableFuture<MessageId> lastSendFuture = CompletableFuture.completedFuture(null);
 
-    // Globally unique producer name
+    /**
+     * Globally unique producer name generated by server. It should be the same as {@link #producerName} unless user
+     * configures {@link ProducerConfigurationData::setProducerName}.
```

Comment:
This will always be the same as `producerName`. If the client overrides it, the server will use that name, but if a producer with same name is already connected, it will instead error it out

Reply:
I have changed it but I don't think `user_producer_id` + `sequence_id` can create UUID because `user_producer_id` can be reused by other producer later on and `sequence_id` can start with `0` again and it will create a duplicate combination which will not be unique across messages on the topic and that can be issue for message chunking.
eg:
Producer P1 gives name `userProducer1` and publish message with sequenceId=`0`. Now that process died and another process created producer with similar name `userProducer1` and started message with sequenceId=`0` in which case UUID will not be unique.
Therefore, using `serverProducerName` always gives a guarantee for unique producer-id and UUID. Adding new string `serverProducerId` will not create any overhead as well. So, can you please let me know your thought on it.

Reply:
@merlimat did you get chance to read my above comment before merging the PR?